src/Microsoft.OData.Client/Serialization/HttpWebRequestMessage.cs

```diff
@@ -63,149 +63,143 @@ public class HttpWebRequestMessage : DataServiceClientRequestMessage
-                List<KeyValuePair<string, string>> headerValues = new List<KeyValuePair<string, string>>(this.httpRequest.Headers.Count);
-                foreach (string headerName in this.httpRequest.Headers.AllKeys)
-                {
-                    string headerValue = this.httpRequest.Headers[headerName];
-                    headerValues.Add(new KeyValuePair<string, string>(headerName, headerValue));
-                }
-
-                return headerValues;
+                return _handler.Credentials;
             }
-        }
-
-        /// <summary> Returns the underlying HttpWebRequest </summary>
-        public System.Net.HttpWebRequest HttpWebRequest
-        {
-            get
+            set
             {
-                return this.httpRequest;
+                _handler.Credentials = value;
             }
         }
 
-        /// <summary>
-        /// Gets or set the credentials for this request.
-        /// </summary>
-        public override System.Net.ICredentials Credentials
-        {
-            get { return this.httpRequest.Credentials; }
-            set { this.httpRequest.Credentials = value; }
-        }
+
+#if !(NETCOREAPP1_0 || NETCOREAPP2_0)
 
         /// <summary>
         /// Gets or sets the timeout (in seconds) for this request.
         /// </summary>
         public override int Timeout
         {
-            get { return this.httpRequest.Timeout; }
-            set { this.httpRequest.Timeout = (int)Math.Min(Int32.MaxValue, new TimeSpan(0, 0, value).TotalMilliseconds); }
+            get
+            {
+                return (int)_client.Timeout.TotalSeconds;
+            }
+            set
+            {
+                _client.Timeout = new TimeSpan(0, 0, value);
+            }
         }
 
         /// <summary>
-        /// Gets or sets the read and write timeout (in seconds) for this request.
+        /// Gets or sets the read write timeout (in seconds) for this request.
+        /// setting the value of the read write timeout to be similar to that of timeout value. 
+        /// this property will not be used as it is not available in HttpClient and getting rid of it 
+        /// will break people who have been using it before. 
         /// </summary>
         public override int ReadWriteTimeout
         {
-            get { return this.httpRequest.ReadWriteTimeout; }
-            set { this.httpRequest.ReadWriteTimeout = (int)Math.Min(Int32.MaxValue, new TimeSpan(0, 0, value).TotalMilliseconds); }
+            get
+            {
+                return (int)_client.Timeout.TotalSeconds;
+            }
+            set
+            {
+                _client.Timeout = new TimeSpan(0, 0, value);
+            }
         }
 
         /// <summary>
-        /// Gets or sets a value that indicates whether to send data in segments to the
-        ///  Internet resource.
+        /// Gets or sets a value that indicates whether to send data in segments to the Internet resource. 

```

Comment:
```suggestion
        /// Gets or sets a value that indicates whether to send data in segments. 

```

Reply:
why are you suggesting we remove to the internet resource? How does it describe where the data is being sent to?

Reply:
I'd take a guess that since this is a `HttpWebRequestMessage` class, it's implied. Could also be that the service that the request is being sent to is not on the Internet

Reply:
I think saying that the data is sent to an "internet resource" is a bit limiting? But it's a suggestion not a request for change so if you're happy with it then I'm happy too